src/components/TimeSlicing.vue

```diff
@@ -0,0 +1,263 @@
+        return networkToReturn;
+      }
+      return networkToReturn;
+    }
+
+    function exportNetwork() {
+      if (network.value === null) {
+        return;
+      }
+
+      const networkToExport = sliceNetwork();
+      const a = document.createElement('a');
+      a.href = URL.createObjectURL(
+        new Blob([JSON.stringify(networkToExport)], {
+          type: 'text/json',
+        }),
+      );
+      a.download = `${store.state.networkName}_${timeSliceNumber.value}-slices.json`;
+      a.click();
+    }
+
+    return {
+      showOptions,
+      sliceRules,
+      edgeVariables,
+      startTimeVar,
+      endTimeVar,
+      timeSliceNumber,
+      cleanedEdgeVariables,
+      sliceNetwork,
+      exportNetwork,
+      timeRange,
+      timeMin,
+      timeMax,
+      selectedRange,
+    };
+  },
+});
+</script>
+
+<template>
+  <div id="timeslicing">
+    <v-list class="pa-0">
+      <v-subheader class="grey darken-3 py-0 pr-0 white--text">
+        Time Slicing
+
+        <v-spacer />
+
+        <v-btn
+          :min-width="40"
+          :height="48"
+          depressed
+          tile
+          :class="showOptions? `grey darken-2 pa-0` : `grey darken-3 pa-0`"
+          @click="showOptions = !showOptions"
+        >
+          <v-icon color="white">
+            mdi-cog
+          </v-icon>
+        </v-btn>
+      </v-subheader>
+
+      <v-card
+        v-if="showOptions"
+        flat
+        color="white"
+        class="pb-4 pt-2"
+      >
+        <v-list-item>
+          <v-select
+            v-model="startTimeVar"
+            label="Start Time"
+            :items="Array.from(cleanedEdgeVariables)"
+            :hide-details="true"
+            class="mt-3"
+            clearable
+            outlined
+            dense
+          />
+        </v-list-item>
+        <v-list-item>
+          <v-select
+            v-model="endTimeVar"
+            label="End Time"
+            :items="Array.from(cleanedEdgeVariables)"
+            :hide-details="true"
+            class="mt-3"
+            clearable
+            outlined
+            dense
+          />
+        </v-list-item>
+        <v-list-item>
+          <v-range-slider
+            v-model="selectedRange"
+            dense
+            hide-details
+            :min="timeMin"
+            :max="timeMax"
+            thumb-label
+          />
+        </v-list-item>
+
+        <v-list-item>
+          <v-text-field
+            v-model="timeSliceNumber"
+            label="Time Slices"
+            :rules="[sliceRules]"
+            :hide-details="true"
+            class="mt-3"
+            clearable
+            outlined
+            dense
+          />
+        </v-list-item>
+
+        <v-list-item>
+          <v-row>
+            <v-col>
+              <v-btn
+                color="primary"
+                block
+                depressed
+                @click="sliceNetwork"
+              >
+                Generate Slices
```

Comment:
Is generating the slices a manual process? Should we instead have it trigger on page load or when we fetch the network in the store?

Reply:
It is a manual process